dts/bindings/sensor/nxp,s32k-qdec.yaml

```diff
@@ -0,0 +1,112 @@
+# Copyright 2023 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  Quadrature Decoder driver which processes encoder signals to determine MR-BUGGY3 displacement
+
+  with cooperation of S32K344 - eMIOS, TRGMUX and LCU modules.
+
+  The sensor qdec application can be used for testing this driver.
+
+  The following example uses TRGMUX IN2 and IN3 to connect to LCU1 LC0 I0 and I1.
+
+  LCU1 LC0 O2 and O3 connect to eMIOS0 CH6(Clockwise rotation) and
+
+  CH7(Counter Clockwise rotation) via TRGMUX_INT_OUT37 and TRGMUX_INT_OUT38
+
+  micro-ticks-per-rev is set as per BUGGY3 gearbox reduction.
+
+  lcu output filters are set to capture maximum speed sensitivity and avoid channel noise.
+
+   qdec0 {
+        compatible = "nxp,qdec-s32k";
+        pinctrl-0 = <&qdec_s32k>;
+        pinctrl-names = "default";
+        micro-ticks-per-rev = <685440000>;
+        status = "okay";
+        trgmux-inst = <0>;
+        trgmux-io-config =
+            <0 TRGMUX_IP_OUTPUT_EMIOS0_CH5_9_IPP_IND_CH6 TRGMUX_IP_INPUT_LCU1_LC0_OUT_I2>,
+            <1 TRGMUX_IP_OUTPUT_EMIOS0_CH5_9_IPP_IND_CH7 TRGMUX_IP_INPUT_LCU1_LC0_OUT_I3>,
+            <2 TRGMUX_IP_OUTPUT_LCU1_0_INP_I0            TRGMUX_IP_INPUT_SIUL2_IN2>,
+            <3 TRGMUX_IP_OUTPUT_LCU1_0_INP_I1            TRGMUX_IP_INPUT_SIUL2_IN3>;
+        trgmux-maps = < 54 3 >, /* TRGMUX_OUT_37 <-> eMIOS0_CH6 */
+           < 55 4 >; /* TRGMUX_OUT_38 <-> eMIOS0_CH7 */
+        lcu_inst = <1>;
+        lcu-output-filter-config =
+            /* LCU Out HW ID, Rise Filter, Fall Filter */
+            <0 5 5>, /* LCU O0 */
+            <1 5 5>, /* LCU O1 */
+            <2 2 2>, /* LCU O2 */
+            <3 2 2>; /* LCU O3 */
+        emios_inst = <0>;
+        emios-channels = <6 7>;
+   };
+
+compatible: "nxp,qdec-s32k"
+
+include: [pinctrl-device.yaml, sensor-device.yaml]
+
+properties:
+  micro-ticks-per-rev:
+    type: int
+    description: |
+      This is a number that is used to determine how many revolutions * 1000000
+      were done based on the current counter's value.
+
+  trgmux-inst:
+    type: int
+    enum: [0, 1]
+    description: |
+      This gives the hardware instance of TRGMUX module.
+
+  trgmux-io-config:
+    type: array
+    description: |
+      This gives the logic triggers configuration of TRGMUX module.
+      It contains 3 values for each of the 4 logic triggers used:
+        logic trigger number, output, input.
+      Hence, it's length should be '12'.
+      Ex:
+        trgmux-io-config =
+          <0 TRGMUX_IP_OUTPUT_EMIOS0_CH5_9_IPP_IND_CH6 TRGMUX_IP_INPUT_LCU1_LC0_OUT_I2>,
+          <1 TRGMUX_IP_OUTPUT_EMIOS0_CH5_9_IPP_IND_CH7 TRGMUX_IP_INPUT_LCU1_LC0_OUT_I3>,
+          <2 TRGMUX_IP_OUTPUT_LCU1_0_INP_I0            TRGMUX_IP_INPUT_SIUL2_IN2>,
+          <3 TRGMUX_IP_OUTPUT_LCU1_0_INP_I1            TRGMUX_IP_INPUT_SIUL2_IN3>;
+
+  lcu_inst:
+    type: int
+    description: |
+      This gives the hardware instance of LCU module.
+
+  emios_inst:
```

Comment:
The common pattern is to have a phandle to the eMIOS device (e.g. `nxp,emios`) and use a device API to retrieve the hw instance number. The same applies for `lcu_inst` and `trgmux-inst` properties above.
There is already a generic eMIOS driver (https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/misc/nxp_s32_emios/nxp_s32_emios.c) that could be extended for that purpose.

Reply:
If I understand these 2 drivers - nxp_s32_emios.c and pwm_nxp_s32_emios.c drivers correctly then, to be able to extract the handle of eMIOS your driver's DT node should be correlated to the eMIOS DT node which has the REG ADDR of an eMIOS instance which can be compared with the IP_EMIOS_##idx##_BASE in HAL and added in a list using LISTIFY.

But my DT node is a QDEC node which has LCU, TRGMUX and eMIOS, which can't be a direct child of eMIOS.
But I believe we can separate out these drivers and DT nodes.. we can discuss about this in detail

Reply:
Yep, but i think is also relates to what Manuel suggested to breakdown the current qdec driver.(https://github.com/zephyrproject-rtos/zephyr/pull/65035#pullrequestreview-1723740577)

Reply:
fixed for eMIOS.. but for lcu and trgmux I would need separate DT nodes for referring to their phandles, which will be possible only once I separate them out

Reply:
Added phandles for lcu and trgmux